Docs deploy fails: `npm install` mutates package-lock.json, blocks gh-pages checkout

## Problem

CI workflow `Deploy Documentation` (`.github/workflows/docs-deploy.yml`) fails at step `Checkout gh-pages branch`:

```
Run git fetch origin gh-pages:gh-pages 2>/dev/null || true
error: Your local changes to the following files would be overwritten by checkout:
    docs/package-lock.json
Please commit your changes or stash them before you switch branches.
Aborting
Error: Process completed with exit code 1.
```

## Root cause

Step `Install dependencies` (line 106-108) runs `npm install`, which can mutate `docs/package-lock.json` (refreshes integrity hashes, adjusts metadata). The subsequent `git checkout gh-pages` (line 123) then refuses because the working tree is dirty.

## Fix

Use `npm ci` instead of `npm install` in the deploy workflow. `npm ci`:
- Reads from `package-lock.json` strictly, never mutates it
- Fails fast on lockfile/`package.json` drift (which is the correct behavior for CI)
- Is the idiomatic choice for CI environments

The lockfile is already maintained by Renovate (`:maintainLockFilesWeekly` preset in `renovate.json`), so drift is kept low.

## Location

`.github/workflows/docs-deploy.yml:106-108`:

```yaml
- name: Install dependencies
  run: npm install
  working-directory: docs
```

→

```yaml
- name: Install dependencies
  run: npm ci
  working-directory: docs
```

## Impact

Blocks docs deploy on every push to `main` that touches `docs/**`.

--- a/.github/workflows/docs-deploy.yml
+++ b/.github/workflows/docs-deploy.yml
@@ -104,7 +104,7 @@ jobs:
           cache-dependency-path: 'docs/package-lock.json'
 
       - name: Install dependencies
-        run: npm install
+        run: npm ci
         working-directory: docs
 
       - name: Build documentation

